components/authz-service/storage/postgres/migration/sql/52_add_db_id_members.up.sql

```diff
@@ -0,0 +1,156 @@
+BEGIN;
+ALTER TABLE iam_policy_members
+    DROP CONSTRAINT iam_policy_members_member_id_fkey;
+ALTER TABLE iam_members
+    DROP CONSTRAINT iam_members_pkey;
+ALTER TABLE iam_members
+    ADD COLUMN db_id SERIAL PRIMARY KEY UNIQUE;
+ALTER TABLE iam_policy_members RENAME COLUMN member_id TO member_temp_id;
+ALTER TABLE iam_policy_members
+    ADD COLUMN member_id INTEGER REFERENCES iam_members (db_id) ON DELETE CASCADE;
+UPDATE
+    iam_policy_members ipm
+SET
+    member_id = (
+        SELECT
+            db_id
+        FROM
+            iam_members
+        WHERE
+            id = ipm.member_temp_id);
+ALTER TABLE iam_policy_members
+    DROP COLUMN member_temp_id;
+ALTER TABLE iam_members
+    ADD UNIQUE (id, name);
+ALTER TABLE iam_policy_members
+    ADD UNIQUE (member_id, policy_id);
+-- add helper for member db_id lookup
+CREATE FUNCTION member_db_id (_id UUID)
+    RETURNS INTEGER
+    AS $$
+    SELECT
+        db_id
+    FROM
+        iam_members
+    WHERE
+        id = _id;
+$$
+LANGUAGE SQL;
+CREATE OR REPLACE FUNCTION projects_match (_projects TEXT[], _projects_filter TEXT[])
+    RETURNS BOOLEAN
+    AS $$
+BEGIN
+    RETURN (
+        -- no projects filter requested (length 0) will be the case for v2.0 or v2.1 ["*"]
+        array_length(_projects_filter, 1) IS NULL
+        -- projects filter intersects with projects for row
+        OR _projects && _projects_filter
+        -- projects for row is an empty array, check if (unassigned) in project filter
+        OR (array_length(_projects, 1) IS NULL
+            AND '{(unassigned)}' && _projects_filter));
+END
+$$
+LANGUAGE PLPGSQL;
+CREATE OR REPLACE FUNCTION query_policy (_policy_id TEXT, _projects_filter TEXT[])
+    RETURNS json
+    AS $$
+    WITH temp AS (
+        SELECT
+            pol.id,
+            pol.name,
+            pol.type,
+            -- get policy's statements using temporary table
+            ( WITH statement_rows AS (
+                    SELECT
+                        stmt.id,
+                        stmt.effect,
+                        stmt.actions,
+                        stmt.resources,
+                        stmt.role,
+                        -- get each statement's projects by cross-referencing iam_policy_statements and iam_statement_projects
+                        (
+                            SELECT
+                                COALESCE(Json_agg(proj.id) FILTER (WHERE proj.id IS NOT NULL),
+                                    '[]')
+                                FROM iam_statement_projects AS stmt_projs
+                            LEFT OUTER JOIN iam_projects AS proj ON stmt_projs.statement_id = stmt.id WHERE stmt_projs.project_id = proj.id) AS projects FROM iam_policy_statements AS pol_stmts
+                    LEFT OUTER JOIN iam_statements AS stmt ON pol_stmts.statement_id = stmt.id WHERE pol_stmts.policy_id = pol.id
```

Comment:
The formatting changes particularly on these couple lines above make it very difficult and error prone to read.